src/parquet/statistics.cc

```diff
@@ -96,6 +96,67 @@ void TypedRowGroupStatistics<DType>::Reset() {
   has_min_max_ = false;
 }
 
+template <typename T>
+inline int getValueBeginOffset(const T* values, int64_t count) {
+  return 0;
+}
+
+template <typename T>
+inline int getValueEndOffset(const T* values, int64_t count) {
+  return count;
+}
+
+template <typename T>
+inline bool notNaN (const T* value) {
+  return true;
+}
+
+template <>
+inline int getValueBeginOffset<float>(const float* values, int64_t count) {
+  // Skip NaNs
+  for (int64_t i = 0; i < count; i++) {
+     if (!std::isnan(values[i])) return i;
+  }
+  return count;
+}
+
+template <>
+inline int getValueEndOffset<float>(const float* values, int64_t count) {
+  // Skip NaNs
+  for (int64_t i = (count - 1); i > 0; i--) {
+     if (!std::isnan(values[i])) return (i + 1);
+  }
+  return count;
+}
+
+template <>
+inline bool notNaN<float>(const float* value) {
+  return !std::isnan(*value);
+}
+
+template <>
+inline int getValueBeginOffset<double>(const double* values, int64_t count) {
+  // Skip NaNs
+  for (int64_t i = 0; i < count; i++) {
+     if (!std::isnan(values[i])) return i;
+  }
+  return 0;
+}
+
+template <>
+inline int getValueEndOffset<double>(const double* values, int64_t count) {
+  // Skip NaNs
+  for (int64_t i = (count - 1); i > 0; i--) {
```

Comment:
I think it should be `i >= 0` instead of `i > 0`
Think of the case when only the first element is a number, e.g.:
`{3.14}`, or
`{3.14, NaN, NaN, NaN, NaN, ..., NaN}`
For these inputs, this function will return 0.
`getValueBeginOffset()` will also return 0.
Usually, C++ ranges are interpreted as `[first, last)`, ie. they are open at the end. Therefore, `[0, 0)` is an empty range.
However, this solution will work, because at L178 you don't return when `begin_offset_ == end_offset`, and `minmax_element()` returns `make_pair(first, first)` if the range is empty.

But I feel like it currently works by accident, and would be better to fix this.

Reply:
Fixed!